extensions/tls-registry/cli/src/main/java/io/quarkus/tls/cli/acme/AcmeActions.java

```diff
@@ -0,0 +1,231 @@
+package io.quarkus.tls.cli.acme;
+
+import static java.lang.System.Logger.Level.INFO;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.security.KeyFactory;
+import java.security.PrivateKey;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.Base64;
+
+import org.wildfly.security.x500.cert.X509CertificateChainAndSigningKey;
+import org.wildfly.security.x500.cert.acme.AcmeAccount;
+import org.wildfly.security.x500.cert.acme.AcmeException;
+
+import io.quarkus.tls.cli.helpers.LetsEncryptHelpers;
+import io.vertx.core.json.JsonObject;
+
+public final class AcmeActions {
+    static System.Logger LOGGER = System.getLogger("lets-encrypt-actions");
+
+    public static void prepare(String letsEncryptFolder) {
+        try {
+            LOGGER.log(INFO, "Creating a working directory");
+            Files.createDirectories(Paths.get(letsEncryptFolder));
+        } catch (IOException ex) {
+            LOGGER.log(INFO, "Failed to create a working directory");
+            throw new RuntimeException(ex);
+        }
+    }
+
+    public static String createAccount(AcmeClient acmeClient,
+            String letsEncryptPath,
+            boolean staging,
+            String contactEmail) {
+        prepare(letsEncryptPath);
+
+        LOGGER.log(INFO, "Creating {0} Let's Encrypt account", (staging ? "staging" : ""));
+
+        AcmeAccount acmeAccount = AcmeAccount.builder().setTermsOfServiceAgreed(true)
+                .setServerUrl("https://acme-v02.api.letsencrypt.org/directory")
+                .setStagingServerUrl("https://acme-staging-v02.api.letsencrypt.org/directory")
+                .setContactUrls(new String[] { "mailto:" + contactEmail }).build();
+        try {
+            if (!acmeClient.createAccount(acmeAccount, staging)) {
+                LOGGER.log(INFO, "{0} Let's Encrypt account {1} already exists", (staging ? "Staging" : ""), contactEmail);
+            } else {
+                LOGGER.log(INFO, "{0} Let's Encrypt account {1} has been created", (staging ? "Staging" : ""), contactEmail);
+            }
+        } catch (AcmeException ex) {
+            LOGGER.log(INFO, "Failed to create account");
+            throw new RuntimeException(ex);
+        }
+        JsonObject accountJson = convertAccountToJson(acmeAccount);
+        saveAccount(letsEncryptPath, accountJson);
+        return accountJson.encode();
+    }
+
+    public static String firstCertificate(AcmeClient acmeClient,
+            String letsEncryptPath,
+            boolean staging,
+            String domain,
+            String certChainPemLoc,
```

Comment:
Here if you'd like we can pass a pem root folder and expected cert and key pem file names and calculate the complete path inside of this function